diff --git a/node_modules/react-native-gesture-handler/apple/RNGestureHandlerModule.mm b/node_modules/react-native-gesture-handler/apple/RNGestureHandlerModule.mm
index 594eba3..16f6290 100644
--- a/node_modules/react-native-gesture-handler/apple/RNGestureHandlerModule.mm
+++ b/node_modules/react-native-gesture-handler/apple/RNGestureHandlerModule.mm
@@ -101,19 +101,13 @@ void decorateRuntime(jsi::Runtime &runtime)
         if (!arguments[0].isObject()) {
           return jsi::Value::null();
         }
-        auto shadowNode = shadowNodeFromValue(runtime, arguments[0]);
-
-        if (dynamic_pointer_cast<const ParagraphShadowNode>(shadowNode)) {
-          return jsi::Value(true);
-        }
-
-        if (dynamic_pointer_cast<const TextShadowNode>(shadowNode)) {
-          return jsi::Value(true);
+        auto shadowNodeList = shadowNodeListFromValue(runtime, arguments[0]);
+        if (!shadowNodeList.empty()) {
+          auto shadowNode = shadowNodeList[0];
+          bool isViewFlatteningDisabled = shadowNode->getTraits().check(ShadowNodeTraits::FormsStackingContext);
+          return jsi::Value(isViewFlatteningDisabled);
         }
-
-        bool isViewFlatteningDisabled = shadowNode->getTraits().check(ShadowNodeTraits::FormsStackingContext);
-
-        return jsi::Value(isViewFlatteningDisabled);
+        return jsi::Value::null();
       });
   runtime.global().setProperty(runtime, "isViewFlatteningDisabled", std::move(isViewFlatteningDisabled));
 }
